.github/workflows/tests.yaml

```diff
@@ -0,0 +1,116 @@
+name: Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  Linting:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - uses: pre-commit/action@v2.0.0
+
+  Test:
+    needs: Linting
+    name: ${{ matrix.os }}, ${{ matrix.env }}, POSTGIS=${{ matrix.postgis }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        postgis: [false]
+        dev: [false]
+        env:
+          - ci/travis/36-minimal.yaml
+          - ci/travis/38-no-optional-deps.yaml
+          - ci/travis/36-pd025.yaml
+          - ci/travis/37-latest-defaults.yaml
+          - ci/travis/37-latest-conda-forge.yaml
+          - ci/travis/38-latest-conda-forge.yaml
+        include:
+          - env: ci/travis/37-latest-conda-forge.yaml
+            os: macos-latest
+            postgis: false
+            dev: false
+          - env: ci/travis/38-latest-conda-forge.yaml
+            os: macos-latest
+            postgis: false
+            dev: false
+          - env: ci/travis/37-latest-conda-forge.yaml
+            os: windows-latest
+            postgis: false
+            dev: false
+          - env: ci/travis/38-latest-conda-forge.yaml
+            os: windows-latest
+            postgis: false
+            dev: false
+          - env: ci/travis/38-latest-conda-forge.yaml
+            os: ubuntu-latest
+            postgis: true
+            dev: false
+          - env: ci/travis/37-dev.yaml
+            os: ubuntu-latest
+            postgis: false
+            dev: true
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Conda Env
+        uses: goanpeca/setup-miniconda@v1
+        with:
+          auto-update-conda: true
+          auto-activate-base: false
+          activate-environment: test
+          environment-file: ${{ matrix.env }}
+          channel-priority: strict
+
+      - name: Check and Log Environment
+        shell: bash -l {0}
+        run: |
+          python -V
+          conda info
+          conda list 2>&1 | tee conda.txt
+          if ( cat conda.txt | grep -q pygeos  )
+          then
+            echo "Setting HAS_PYGEOS=1"
+            echo '::set-env name=HAS_PYGEOS::1'
+          else
+            echo "Setting HAS_PYGEOS=0"
+            echo '::set-env name=HAS_PYGEOS::0'
+          fi
+
+      - name: Test without PyGEOS
+        shell: bash -l {0}
+        continue-on-error: ${{ matrix.dev }}
+        env:
+          USE_PYGEOS: 0
+        run: |
+          pytest -v -r s -n auto --color=yes --cov=geopandas --cov-append --cov-report term-missing --cov-report xml geopandas/
```

Comment:
Above you mentioned at some point that it doesn't actually run faster on CI with pytest-xdist? If that's the case, I would just leave out the `-n auto` (we can still recommend it for local usage in the contributing docs, where it makes an actual difference)

Reply:
Here it does help. But not too much since most of the time is spent in conda anyway.

Reply:
> Here it does help. But not too much since most of the time is spent in conda anyway.

Do you know how much? (it halves the time spent in the tests?) 
Just asking, because if it is not much, I would maybe rather leave it out to avoid additional complexity (or strange issues from running tests in parallel). In which case we can of course still recommend it for local use in the contributing guide